src/cli/repl/index.js

```diff
@@ -35,22 +46,21 @@ function prettyPrint(text, o, depth) {
 
 // This lets us handle promises more gracefully than the default REPL,
 // which doesn't show the results.
-function promiseFriendlyWriter(displayPrompt) {
-  const PRINT_DEPTH = 15;
-  return (result) => promisePrint(result, displayPrompt, PRINT_DEPTH);
+function promiseFriendlyWriter({ displayPrompt, getPrintDepth }) {
+  return (result) => promisePrint(result, displayPrompt, getPrintDepth);
 }
 
-function promisePrint(result, displayPrompt, depth) {
+function promisePrint(result, displayPrompt, getPrintDepth) {
   if (result && typeof result.then === 'function') {
     // Bit of a hack to encourage the user to wait for the result of a promise
     // by printing text out beside the current prompt.
     Promise.resolve()
       .then(() => console.log('Waiting for promise...'))
       .then(() => result)
-      .then((o) => prettyPrint('Promise Resolved: \n', o, depth))
-      .catch((err) => prettyPrint('Promise Rejected: \n', err, depth))
+      .then((o) => prettyPrint('Promise Resolved: \n', o, getPrintDepth()))
```

Comment:
I feel like we should be capturing the print depth when we receive the result, not reading it once it resolves.